src/runtimes.ts

```diff
@@ -0,0 +1,63 @@
+/* eslint-disable no-var */
+declare global {
+  var EdgeRuntime: any;
+  var Netlify: any;
+  var Bun: any;
+  var Deno: any;
+  var __lagon__: any;
+  var fastly: any;
+}
+
+// https://runtime-keys.proposal.wintercg.org/
+export type Runtime =
+  | "workerd"
+  | "deno"
+  | "lagon"
+  | "netlify"
+  | "node"
+  | "bun"
+  | "edge-light"
+  | "fastly"
+  | "";
+
+// https://developers.cloudflare.com/workers/runtime-apis/web-standards/#navigatoruseragent
+const CLOUDFLARE_WORKERS_NAVIGATOR = "Cloudflare-Workers";
+
+// https://nodejs.org/api/process.html#processrelease
+const NODE_PROCESS_RELEASE_NAME = "node";
+
+export function detectRuntime(): Runtime {
```

Comment:
I would suggest to introduce `Runtime: { name: RuntimeName }` interface for future compatibility